### ISSUE TITLE
Harden `isFile` method

### ISSUE THREAD

**Issue by marksmccann** [marksmccann/node-sass-extra#46]
It's probably a good idea to harden the `isFile` method in our package. Currently, it is simply checking for a file extension and assuming it is a file, but a directory can have a file extension as well (e.g. 'path/to/dir.css/file.css'). We should probably use `stats.isFile()` and/or `stats.isDirectory()` to harden this check, at least for determining the source files.

**Comment by marksmccann** [comment]
@phillipluther Do you think if we did this, it would solve some of our problems with the CLI? I mean we can more accurately determine if the user is referring to a directory or a file for their source. If it is a directory, we can handle it the same way as the CLI.

**Comment by phillipluther** [comment]
@marksmccann Without an explicit flag that says, "HEY! I want this file-looking thing to be a directory!" I'm not sure it would help - I mean, if somebody legitimately wants to have a directory called `styles.css` ... I'm not sure how we'd be able to determine that from a simple string. 

If it's in a build and that directory doesn't exist yet, we couldn't check against it. Maybe there's some combination of other settings that would determine if the intent is dir or file?

**Comment by marksmccann** [comment]
@phillipluther Maybe it is my ignorance, but I was under the impression that `stats.isFile()` and `stats.isDirectory()` actually check the file tree to see if the given path is either a file or a directory in actuality, not just looking at the string. If so, wouldn't that work?

**Comment by marksmccann** [comment]
I haven't tested it yet, but after a quick search, I think that my assumption is correct. If we can accurately determine that the source is a directory, we can then search for sass files inside of it, right?

**Comment by phillipluther** [comment]
Yes, but when that string comes in as an option it might not exist yet. We’re doing an ‘ensureDir’ to create the path if it doesn’t exist, aren’t we?

I’ll double check that, too. If not, you’re totally right.

________________________________

**Comment by marksmccann** [comment]
For the output, I think that you are right, but the source should always be defined. I guess this solves half the problem :). I guess for output we can just make an assumption and then document it?
